src/plugins/event-plugins/DomEventPlugin.ts

```diff
@@ -132,20 +135,51 @@ export class DomEventPlugin<
 
     private getEventListener(cssLocator?: string): EventListener {
         return (event: Event): void => {
+            const elementId = this.getElementId(event);
+            let customId: string;
+            if (this.config.customIdTracker) {
+                customId = this.config.customIdTracker(event);
+            }
+
             const eventData: DomEvent = {
                 version: '1.0.0',
                 event: event.type,
-                elementId: DomEventPlugin.getElementId(event)
+                element: this.getElement(event),
+                ...(elementId ? { elementId } : {}),
+                ...(customId ? { customId } : {}),
+                ...(cssLocator ? { cssLocator } : {})
             };
-            if (cssLocator !== undefined) {
-                eventData.cssLocator = cssLocator;
-            }
+
             if (this.context?.record) {
                 this.context.record(DOM_EVENT_TYPE, eventData);
             }
         };
     }
 
+    private getElement(event: Event): string {
+        if (!event.target) {
+            return '';
+        }
+
+        if (event.target instanceof Node) {
+            return event.target.nodeName;
+        }
+
+        return '';
+    }
+
+    private getElementId(event: Event): string {
```

Comment:
issue: I think we should update the function at line 69 rather than creating a new one (and move `getElement` to be co-located with the function at line 69).

Reply:
Sounds good. Will add this.